test/Feature/HLSLLib/PartiallyMappedResources.test

```diff
@@ -0,0 +1,132 @@
+#--- source.hlsl
+
+struct S{
+  int data[512];
+};
+
+StructuredBuffer<S> X : register(t0);
+StructuredBuffer<S> Y : register(t1);
+
+RWStructuredBuffer<int> Out : register(u2);
+RWStructuredBuffer<int> CAFM : register(u3);
+
+
+[numthreads(1,1,1)]
+void main() {
+  // 32 S structs inside X or Y occupy 64KB of data. (32 * 512 ints * 4 bytes per int)
+  // So, any index into the buffer >= [32] will access a new "tile"
+
+  int idx = 0;
+
+  uint status;
+  S Out0 = X.Load(0, status);
+  Out[idx] = Out0.data[0];
+  Out[idx + 4] = status;
+  CAFM[idx] = CheckAccessFullyMapped(status) ? 1 : 0;
+  idx += 1;
+
+  S Out1 = X.Load(50, status);
+  Out[idx] = Out1.data[0];
+  Out[idx + 4] = status;
+  CAFM[idx] = CheckAccessFullyMapped(status) ? 1 : 0;
+  idx += 1;
+
+  S Out2 = Y.Load(0, status);
+  Out[idx] = Out2.data[0];
+  Out[idx + 4] = status;
+  CAFM[idx] = CheckAccessFullyMapped(status) ? 1 : 0;
+  idx += 1;
+
+  S Out3 = Y.Load(50, status);
+  Out[idx] = Out3.data[0];
+  Out[idx + 4] = status;
+  CAFM[idx] = CheckAccessFullyMapped(status) ? 1 : 0;
+}
+//--- pipeline.yaml
+
+---
+Shaders:
+  - Stage: Compute
+    Entry: main
+    DispatchSize: [1, 1, 1]
+Buffers:
+  - Name: X
+    Format: Int32
+    Stride: 2048 # S is 512 ints, 512*4 = 2048.
+    FillSize: 131072
+    FillValue: 1
+  - Name: Y
+    Format: Int32
+    Stride: 2048
+    FillSize: 131072
+    FillValue: 2
+  - Name: Out
+    Format: Int32
+    Stride: 4
+    FillSize: 32
+  - Name: ExpectedOut
+    Format: Int32
+    Stride: 4
+    # first 4 values are the actual data retrieved. For non-resident loads, 0 is expected.
+    # last 4 values are the status. 1 is expected for resident memory, 0 for non-resident
+    Data: [1, 0, 0, 0, 1, 0, 0, 0]
+  - Name: CAFM
+    Format: Int32
+    Stride: 4
+    FillSize: 16
+    FillValue: 0
+  - Name: ExpectedCAFM
+    Format: Int32
+    Stride: 4
+    # Only the first data access should be accessing fully mapped memory
+    Data: [1, 0, 0, 0]
+
+Results:
+  - Result: Test
+    Rule: BufferExact
+    Actual: Out
+    Expected: ExpectedOut
+  - Result: TestCAFM
+    Rule: BufferExact
+    Actual: CAFM
+    Expected: ExpectedCAFM
+DescriptorSets:
+  - Resources:
+    - Name: X
+      Kind: StructuredBuffer
+      DirectXBinding:
+        Register: 0
+        Space: 0
+      VulkanBinding:
+        Binding: 0
+      TilesMapped: 1

```

Comment:
Are we sure `TilesMapped` really belongs on the DescriptorSets part here?  I would expect it to be a part of the Buffers.

My reasoning here is that it becomes ambiguous what to do if multiple descriptors refer to the same resource but have different TilesMapped settings on them.

Reply:
Oh, actually it looks like maybe the current structure of this creates the descriptors at the same time as the resources, so even though it doesn't look like it they are actually tied together....I keep being surprised that this is how this code works, but fixing that is way beyond the scope of this PR.

However, I do still think it makes sense to describe how the resource is mapped into memory in the Buffers section rather than the DescriptorSets one.